### ISSUE TITLE
$__timeGroup macro grouping dates wrong on Sql Server

### ISSUE THREAD

**Issue by philippk80** [grafana/grafana#23765]
**What happened**:
We tried to group by month with the $__timeGroup feature and one day we discovered that rows are missing to a month. Possibly this happens on other months too.

**What you expected to happen**:
We expect the two dates to occur in the same month.

**How to reproduce it (as minimally and precisely as possible)**:
```
SELECT 
    FLOOR(DATEDIFF(second, '1970-01-01', '2020-04-08 01:15:40.0530000 +02:00')/2592000)*2592000
    ,FLOOR(DATEDIFF(second, '1970-01-01', '2020-04-08 19:56:31.3200000 +02:00')/2592000)*2592000;
```
**Anything else we need to know?**:
The date column is originally defined as datetimeoffset(7).
The 2592000 will be generated by Grafana when we select 'This month' or when we select '01.04. to 30.04'

**Environment**:
- Grafana version: 
    v6.4.2
- Data source type & version:
    SQL Server 2014
- OS Grafana is installed on:
    Docker
- User OS & Browser:
    Windows 10
    Chrome
- Grafana plugins:
- Others:

**Comment by marefr** [comment]
Please include full content of query editor here, thanks.

**Comment by philippk80** [comment]
SELECT time = $__timeGroup(DateColumn, $GroupIntervall)
       ,metric = ISNULL(MetricColumn, 'Unknown')
       ,value = COUNT(ID)
FROM [SchemaName].[TableName] WITH(READUNCOMMITTED)
WHERE Status = 10 AND $__timeFilter(DateColumn)
GROUP BY ISNULL(MetricColumn, 'Unknown'), $__timeGroup(DateColumn, $GroupIntervall)

**Comment by marefr** [comment]
Please group by `$__timeGroup(DateColumn, $GroupIntervall)` first, then `ISNULL(MetricColumn, 'Unknown')`. Then add an `ORDER BY $__timeGroup(DateColumn, $GroupIntervall), ISNULL(MetricColumn, 'Unknown')`

Regarding month (M) in interval, that haven't been supported until #23120 was merged which will be included in Grafana v7.0.
